hack/test-end-to-end-docker.sh

```diff
@@ -36,7 +36,7 @@ function cleanup()
 	dump_container_logs
 
 	echo "[INFO] Dumping all resources to ${LOG_DIR}/export_all.json"
-	oc export all --all-namespaces --raw -o json --config=${ADMIN_KUBECONFIG} > ${LOG_DIR}/export_all.json
+	oc export all --all-namespaces --raw -o json --config=${ADMIN_KUBECONFIG-} > ${LOG_DIR}/export_all.json
```

Comment:
This can be called before ADMIN_KUBECONFIG is set up if the failure happens
early enough.

On Thu, Jun 2, 2016 at 9:02 AM, Steve Kuznetsov <notifications@github.com>
wrote:

> In hack/test-end-to-end-docker.sh
> <https://github.com/openshift/origin/pull/9102#discussion_r65534428>:
>
> > @@ -36,7 +36,7 @@ function cleanup()
> >  	dump_container_logs
> >
> >  	echo "[INFO] Dumping all resources to ${LOG_DIR}/export_all.json"
> > -	oc export all --all-namespaces --raw -o json --config=${ADMIN_KUBECONFIG} > ${LOG_DIR}/export_all.json
> > +	oc export all --all-namespaces --raw -o json --config=${ADMIN_KUBECONFIG-} > ${LOG_DIR}/export_all.json
>
> We prefer to use ${var:-default} because it's rare that someone wants the
> set-but-null behavior from ${var-default} and keeping that nuance in mind
> is hard.
>
> Also, why do we expect this export to work if we *don't* have
> $ADMIN_KUBECONFIG? AFAICT, it is *always* set once a cluster has been
> successfully stood up. If one hasn't, would we expect oc export to work?
> If one has, would we expect the output of oc export to be useful without
> the admin credentials?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9102/files/4297fea49f7f647fb18c8a8129f766080532216a#r65534428>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p-HUTQrwzJsc6iz5tytGpsgIt0elks5qHtRmgaJpZM4IrHju>
> .
>